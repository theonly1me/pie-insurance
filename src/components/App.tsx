import { lazy, Suspense } from 'react';
import Loader from './reusable/Loader';

const App: React.FC = () => {
  const Hero = lazy(() => import('./Hero'));
  const Header = lazy(() => import('./Header'));
  const Details = lazy(() => import('./Details'));
  const Modules = lazy(() => import('./Modules'));
  const Footer = lazy(() => import('./Footer'));
  return (
    <>
      <Suspense fallback={<Loader />}>
        <Header />
        <main>
          <Hero />
          <Details />
          <Modules />
        </main>
        <Footer />
      </Suspense>
    </>
  );
};

export default App;
