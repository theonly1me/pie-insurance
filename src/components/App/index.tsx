import { lazy, Suspense } from 'react';
// components
import Loader from '../reusable/Loader';
import Header from '../Header';
import Footer from '../Footer';
import Details from '../Details';
import Hero from '../Hero';

const App: React.FC = () => {
  //Lazy Load
  const Modules = lazy(() => import('../Modules'));

  return (
    <>
      <Header />
      <main>
        <Hero />
        <Details />
        <Suspense fallback={<Loader />}>
          <Modules />
        </Suspense>
      </main>
      <Footer />
    </>
  );
};

export default App;
