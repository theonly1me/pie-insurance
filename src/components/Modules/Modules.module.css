@import '../../globals.css';

.modules {
    padding: 9.6rem 0;
}

