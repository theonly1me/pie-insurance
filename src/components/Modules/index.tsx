// styles
import classes from './Modules.module.css';
// components
import SectionHeading from '../reusable/SectionHeading/SectionHeading';
import Card from '../reusable/Card';
// assets
import bike from '../../assets/bike.svg';
import jewelry from '../../assets/jewelry.svg';
import electronics from '../../assets/electronics.svg';
import sports from '../../assets/sports.svg';

// data
/* Currently stored in a local JSON file, ideally should come from an API */
import moduleData from '../../data/data.json';

/**
 * Modules Component
 */
const Modules = () => {
  const images: { [key: string]: string } = {
    bike,
    jewelry,
    electronics,
    sports,
  };

  const renderCard = () =>
    moduleData.map(data => (
      <Card
        key={data.id}
        id={data.id}
        attributes={data.attributes}
        coverage={data.coverage}
        image={images[data.img]}
        title={data.title}
        range={data.range}
        risk={data.risk}
      />
    ));
  return (
    <section id="modules" className={classes.modules}>
      <SectionHeading
        subheading="Browse through our offerings"
        heading="Modules"
      />
      <div className={`container grid gridCol2`}>{renderCard()}</div>
    </section>
  );
};

export default Modules;
