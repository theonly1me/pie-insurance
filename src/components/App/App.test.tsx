import { act, render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import mockHeader from '../Header';
import mockHero from '../Hero';
import mockDetails from '../Details';
import mockFooter from '../Footer';

import App from './index';
import { Suspense } from 'react';
import Modules from '../Modules';
import Loader from '../reusable/Loader';

describe('Test App Component', () => {
  it('Renders App Component Successfully', async () => {
    render(<App />);
  });

  it('Renders Header Section into the document', async () => {
    jest.mock('../Header', () => {
      return mockHeader;
    });

    render(<App />);
    const component = await screen.findByTestId('header');
    expect(component).toBeInTheDocument();
  });

  it('Renders the Nav section of the header containing the shopping cart button', async () => {
    jest.mock('../Header', () => {
      return mockHeader;
    });

    render(<App />);
    const component = await screen.findByRole('navigation');
    expect(component).toBeInTheDocument();
  });

  it('Renders the Hero component and Anchor Buttons', async () => {
    jest.mock('../Hero', () => {
      return mockHero;
    });
    render(<App />);
    const viewModulesAnchorButton = await screen.findByText(/view modules/i);
    const learnMoreAnchorButtom = await screen.findByText(/learn more â†“/i);
    expect(viewModulesAnchorButton).toBeInTheDocument();
    expect(learnMoreAnchorButtom).toBeInTheDocument();
  });

  it('Renders the Details component', async () => {
    jest.mock('../Details', () => {
      return mockDetails;
    });
    render(<App />);
    const detailsComponent = await screen.findByText(/Get your cover today/i);
    expect(detailsComponent).toBeInTheDocument();
  });

  it('Renders the Modules component lazily', async () => {
    let root;
    act(() => {
      root = create(
        <Suspense fallback={<Loader />}>
          <Modules />
        </Suspense>
      );
    });
    expect(root).toMatchSnapshot();
  });

  it('Renders Footer Component into the document', async () => {
    jest.mock('../Footer', () => {
      return mockFooter;
    });

    render(<App />);
    const component = await screen.findByTestId('footer');
    expect(component).toBeInTheDocument();
  });
});
