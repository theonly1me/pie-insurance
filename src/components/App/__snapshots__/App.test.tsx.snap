// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Component Renders the Modules component lazily 1`] = `
<section
  className="modules"
  id="modules"
>
  <div
    className="container"
  >
    <span
      className="subheading"
    >
      Browse through our offerings
    </span>
    <h2
      className="heading"
    >
      Modules
    </h2>
  </div>
  <div
    className="container grid gridCol2 mediaGrid mediaGridCard"
  >
    <div
      className="module"
    >
      <img
        alt="Bikes"
        className="moduleImg"
        src="bike.svg"
      />
      <span
        className="coverage"
      >
        0 € - 3000 €
      </span>
      <div
        className="cardContent"
      >
        <p
          className="cardTitle"
        >
          Bikes
        </p>
        <span
          className="covered"
        >
          Coverage Includes:
        </span>
        <ul
          className="attributes"
        >
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Theft Coverage
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Accident & Vandalism Coverage
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Total Loss Coverage
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Material Damage Protection
          </li>
        </ul>
        <div
          className="customSelect css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-1s2u09g-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-14el2xx-placeholder"
                id="react-select-2-placeholder"
              >
                Select Coverage...
              </div>
              <div
                className=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className=""
                  disabled={false}
                  id="react-select-2-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="module"
    >
      <img
        alt="Jewelry"
        className="moduleImg"
        src="jewelry.svg"
      />
      <span
        className="coverage"
      >
        500 € - 10000 €
      </span>
      <div
        className="cardContent"
      >
        <p
          className="cardTitle"
        >
          Jewelry
        </p>
        <span
          className="covered"
        >
          Coverage Includes:
        </span>
        <ul
          className="attributes"
        >
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Theft Coverage
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Water, Fire or Lightning Damage
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Vandalism
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Corrosion
          </li>
        </ul>
        <div
          className="customSelect css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-1s2u09g-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-14el2xx-placeholder"
                id="react-select-3-placeholder"
              >
                Select Coverage...
              </div>
              <div
                className=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-3-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className=""
                  disabled={false}
                  id="react-select-3-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="module"
    >
      <img
        alt="Sports Equipment"
        className="moduleImg"
        src="sports.svg"
      />
      <span
        className="coverage"
      >
        0 € - 20000 €
      </span>
      <div
        className="cardContent"
      >
        <p
          className="cardTitle"
        >
          Sports Equipment
        </p>
        <span
          className="covered"
        >
          Coverage Includes:
        </span>
        <ul
          className="attributes"
        >
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Emergency Medical Expenses
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Repatriation
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Equipment Damage
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Theft Coverage
          </li>
        </ul>
        <div
          className="customSelect css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-4-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-1s2u09g-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-14el2xx-placeholder"
                id="react-select-4-placeholder"
              >
                Select Coverage...
              </div>
              <div
                className=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-4-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className=""
                  disabled={false}
                  id="react-select-4-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="module"
    >
      <img
        alt="Electronics"
        className="moduleImg"
        src="electronics.svg"
      />
      <span
        className="coverage"
      >
        500 € - 6000 €
      </span>
      <div
        className="cardContent"
      >
        <p
          className="cardTitle"
        >
          Electronics
        </p>
        <span
          className="covered"
        >
          Coverage Includes:
        </span>
        <ul
          className="attributes"
        >
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Fall Damage
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Water, Fire or Explosion Damage
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Theft Coverage
          </li>
          <li
            className="attribute"
          >
            <span
              className="listItem"
            >
              +
            </span>
            Total Loss Coverage
          </li>
        </ul>
        <div
          className="customSelect css-b62m3t-container"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-5-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-1s2u09g-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-14el2xx-placeholder"
                id="react-select-5-placeholder"
              >
                Select Coverage...
              </div>
              <div
                className=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-5-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className=""
                  disabled={false}
                  id="react-select-5-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
